### DISCUSSION TITLE
Docker tomcat configuration

### DISCUSSION THREAD

**Discussion by ndevilleBE** [FraunhoferIOSB/FROST-Server#1553] [General]
Dear, 

I use the docker image: fraunhoferiosb/frost-server:2.0.4

We are developing a client application to query, over a certain period of time, the whole sensorthings database to find out which datastream has data inside a specific bounding box along with basic statistics (count, min, max).
This means that a lot of requests have to be made in a short amount of time. It starts well but after a while it hangs and then the apache shut down the connection and we have a timeout error. Monitoring the docker resource consumption (with docker stats) it barely moves so I don't think it is stuck there.

I believe that the main bottleneck is the tomcat configuration file server.xml inside the docker image. The max number of threads is limited by default to 150 per connection and I wish to increase it.
What would be the best way to proceed? Keeping in mind that this settings will need to be kept for future update?!
Or is there a docker environment parameter foreseen for this?

Many thanks for your help

**Comment by hylkevds** [comment]
I strongly suspect the issue is in the database, not in Tomcat. If the number of Tomcat threads was an issue, you would be getting "connection refused" errors, not timeouts. Timeouts mean the database can't generate the answer quick enough.

Check if the value of the [persistence.slowQueryThreshold](https://fraunhoferiosb.github.io/FROST-Server/deployment/db_performance.html) parameter is non-zero and you should be seeing log entries stating which queries are slow.
